src/canvasObject.js

```diff
@@ -52,6 +53,7 @@ CanvasObject.prototype = {
             if(previousImageObj){
               viewer.world.removeItem(previousImageObj);
             }
+            document.dispatchEvent(new CustomEvent('detail-tile-source-opened', { 'detail': self.id }));
```

Comment:
I don't think it's a good idea to use the native API for this. As a library, the custom events it provides should be encapsulated. So, for instance, in the "separate renderer" branch, I've attached all the events to an event emitter for the component. This eventemiiter then has to be passed in or mixed in to the component that uses it.